apps/web/src/pages/templates/workflow/SideBar/SelectedStep.tsx

```diff
@@ -0,0 +1,250 @@
+import React, { SetStateAction } from 'react';
+import styled from '@emotion/styled';
+import { ActionIcon, Divider, Stack } from '@mantine/core';
+import { StepTypeEnum } from '@novu/shared';
+import { Button, colors, Text, Title } from '../../../../design-system';
+import { Close } from '../../../../design-system/icons/actions/Close';
+import { getChannel, NodeTypeEnum } from '../../shared/channels';
+import { IForm } from '../../../../components/templates/use-template-controller.hook';
+import { StepActiveSwitch } from '../StepActiveSwitch';
+import { useEnvController } from '../../../../store/use-env-controller';
+import { When } from '../../../../components/utils/When';
+import { PlusCircle, Trash } from '../../../../design-system/icons';
+import { DigestMetadata } from '../DigestMetadata';
+import { DelayMetadata } from '../DelayMetadata';
+import { Filters } from '../../filter/Filters';
+import { ShouldStopOnFailSwitch } from '../ShouldStopOnFailSwitch';
+import { ReplyCallback } from '../ReplyCallback';
+import { FieldErrors } from 'react-hook-form';
+import { NavSection } from '../../../../components/templates/TemplatesSideBar';
+import { StyledNav } from '../WorkflowEditorPage';
+
+const capitalize = (text: string) => {
+  return typeof text !== 'string' ? '' : text.charAt(0).toUpperCase() + text.slice(1);
+};
+
+export function SelectedStep({
+  selectedChannel,
+  setSelectedChannel,
+  setActivePage,
+  steps,
+  activeStep,
+  control,
+  errors,
+  setFilterOpen,
+  colorScheme,
+  isLoading,
+  isUpdateLoading,
+  loadingEditTemplate,
+  isDirty,
+  onDelete,
+  selectedNodeId,
+}: {
+  selectedChannel:
```

Comment:
Won't it work by only specifying `StepTypeEnum` ?

Reply:
Totally right missed the multi-definition after the refactoring.